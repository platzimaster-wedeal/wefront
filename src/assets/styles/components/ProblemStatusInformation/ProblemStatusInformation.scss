@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.problem-status-information {
  display: flex;
  flex-flow: column nowrap;
  position: relative;
  width: 100%;

  &__status-message {
    display: flex;
    flex-flow: column nowrap;
    padding: $space-size-normal 0;
    
  }
}
