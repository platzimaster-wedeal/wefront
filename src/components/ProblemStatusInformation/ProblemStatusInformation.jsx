import React, { useState, useEffect } from "react";
import { useSelector } from 'react-redux'

// Stlyes
import "../../assets/styles/components/ProblemStatusInformation/ProblemStatusInformation.scss";

// Components
import PostHeader from "../PostHeader/PostHeader";
import FormStep from "../FormStep/FormStep";
import ProfileWorkerCard from "../ProfileWorkerCard/ProfileWorkerCard";
import ProblemQualification from "../ProblemQualification/ProblemQualification";
import Button from "../Buttons/Button";
import ModalContainer from "../Modals/ModalContainer";
import Modal from "../Modals/Modal";

const ProblemStatusInformation = ({
 isUser,
 problemStatus = true,
 problemWorker = "@",
 workersApplied = [],
 setInformation,
 onQualificate
}) => {
  
  // const profile = useSelector(state => state.ProfileReducer)

  const [profile, setProfile] = useState({}) 


 // ---------------- Handle Problem Solved
 const [isProblemSolved, setIsProblemSolved] = useState(false);

 const handleProblemSolved = () => setIsProblemSolved(!isProblemSolved);

 // ---------------- Define the status of the problem ----------------
 const RenderStatus = () => {
  if (problemStatus !== "solving")
   return (
    <>
     <span className="problem-status-information__status-message">
      The user {problemWorker} is solving the problem.
     </span>
     {isUser && !isProblemSolved && (
      <Button active onClick={handleProblemSolved}>
       Problem Solved
      </Button>
     )}
    </>
   );

  return (
   <span className="problem-status-information__status-message">
    The problem is still open!
   </span>
  );
 };

 const RenderWorkers = () => {
  return (
   <FormStep title="Workers That Applied">
    {workersApplied && workersApplied.length > 0 ? (
     workersApplied.map((worker, i) => <ProfileWorkerCard key={i} />)
    ) : (
     <span className="problem-status-information__workers-message">
      No workers have come forward yet
     </span>
    )}
   </FormStep>
  );
 };

 return (
  <article className="problem-status-information">
   <PostHeader />
   <FormStep title="Current Status">{RenderStatus()}</FormStep>

   {isUser && !problemStatus && RenderWorkers()}

   {isUser && isProblemSolved && (
    <FormStep title="Qualification Of Worker">
     <ProblemQualification setInformation={setInformation} onQualificate={onQualificate} />
    </FormStep>
   )}
  </article>
 );
};

export default ProblemStatusInformation;
