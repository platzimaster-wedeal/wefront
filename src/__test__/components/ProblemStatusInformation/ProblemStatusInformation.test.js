import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import ProblemStatusInformation from "../../../components/ProblemStatusInformation/ProblemStatusInformation";

describe("Test for the component ProblemStatusInformation :D", () => {
 const ProblemStatusInformationMounted = mount(<ProblemStatusInformation />);
 const ProblemStatusInformationCreated = create(<ProblemStatusInformation />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(ProblemStatusInformationCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(ProblemStatusInformationMounted.length).toEqual(1);
  });

  test("Should the message of current status step be a string", () => {
   expect(
    typeof ProblemStatusInformationMounted.find(
     ".problem-status-information__status-message"
    ).text()
   ).toEqual("string");
  });
  test("Should the message of current status step be a string", () => {
   expect(
    ProblemStatusInformationMounted.find(
     ".problem-status-information__status-message"
    ).text()
   ).toEqual("The user @Alan is solving the problem.");
  });
  test("Should the message of current status step be a string", () => {
   expect(
    typeof mount(<ProblemStatusInformation problemStatus={false} />)
     .find(".problem-status-information__status-message")
     .text()
   ).toEqual("string");
  });
  test("Should the message of current status step be a The problem is still open!", () => {
   expect(
    mount(<ProblemStatusInformation problemStatus={false} />)
     .find(".problem-status-information__status-message")
     .text()
   ).toEqual("The problem is still open!");
  });

  test("Should the message of non worker of the component be a string", () => {
   expect(
    typeof ProblemStatusInformationMounted.find(
     ".problem-status-information__workers-message"
    ).text()
   ).toEqual("string");
  });
  test("Should the message of non worker of the component be a string", () => {
   expect(
    ProblemStatusInformationMounted.find(
     ".problem-status-information__workers-message"
    ).text()
   ).toEqual("No workers have come forward yet");
  });
 });
});
