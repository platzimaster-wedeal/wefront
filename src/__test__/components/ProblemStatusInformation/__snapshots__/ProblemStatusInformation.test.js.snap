// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component ProblemStatusInformation :D Visual tests for the component should the comopnent have the same tree 1`] = `
<article
  className="problem-status-information"
>
  <section
    className="post-header"
  >
    <figure
      className="post-header__picture"
      onClick={[Function]}
    >
      <img
        className="image image__isOnline undefined"
        src={Object {}}
      />
    </figure>
    <div
      className="post-header__user-info"
    >
      <div
        className="post-header__user-info--main"
      >
        <h3
          onClick={[Function]}
        >
          Alan Walks
        </h3>
        <span>
          Has a new problem!
        </span>
      </div>
      <strong
        className="post-header__user-info--profession"
      >
        <span
          className="label label__inactive"
        >
          Engineer
        </span>
      </strong>
    </div>
    <small
      className="post-header__time"
    >
      1hr
    </small>
  </section>
  <div
    className="form-step false "
  >
    <div
      className="form-step__title"
      onClick={[Function]}
    >
      <h2>
        Current Status
      </h2>
      <svg
        fill="currentColor"
        height={30}
        size={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
    <div
      className="form-step__children form-step__children--close"
    >
      <span
        className="problem-status-information__status-message"
      >
        The user 
        @Alan
         is solving the problem.
      </span>
      <button
        className="button button__primary undefined"
        onClick={[Function]}
        type="button"
      >
        Problem Solved
      </button>
    </div>
  </div>
</article>
`;
